docs/wagtail-migrations.md

```diff
@@ -0,0 +1,81 @@
+
+To this end, there are some utility functions in cfgov-refresh that make this easier. Using these utilities, a Django data migration that modifies a StreamField would follow the following format:
+
+```python
+from django.db import migrations
+
+from v1.util.migrations import migrate_page_types_and_fields
+
+
+def forward_mapper(page_or_revision, data):
+    data = dict(data)
+    # Manipulate the stream block data forwards
+    return data
+
+
+def backward_mapper(page_or_revision, data):
+    data = dict(data)
+    # Manipulate the stream block data backwards
+    return data
+
+
+def forwards(apps, schema_editor):
+    page_types_and_fields = [
+        ('myapp', 'MyPage', 'streamfield_name', 'streamblock_type'),
+    ]
+    migrate_page_types_and_fields(apps, 
+                                  page_types_and_fields,
+                                  forward_mapper)
+
+
+def backwards(apps, schema_editor):
+    page_types_and_fields = [
+        ('myapp', 'MyPage', 'streamfield_name', 'streamblock_type'),
+    ]
+    migrate_page_types_and_fields(apps, 
+                                  page_types_and_fields,
+                                  backward_mapper)
+
+
+class Migration(migrations.Migration):
+    dependencies = []
+    operations = [
+        migrations.RunPython(forwards, backwards),
+    ]
+```
+
+### Utility functions
+
+These functions are available in `v1.util.migrations`.
+
+##### `migrate_page_types_and_fields(apps, page_types_and_fields, mapper)`
+
+Migrate the fields of a wagtail page type using the given mapper function. page_types_and_fields should be a list of 4-tuples providing ('app', 'PageType', 'field_name', 'block type').
+
+The mapper function should take `page_or_revision` and the stream block value.
+
+##### `migrate_stream_field(page_or_revision, field_name, block_type, mapper)`
+
+Migrate a block of the type within a StreamField of the name belonging to the page or revision using the mapper function.
+
+The mapper function should take `page_or_revision` and the stream block value.
+
+##### `get_stream_data(page_or_revision, field_name)`
+
+Get the stream field data for a given field name on a page or a revision.
+
+This function will return a list of `dict`-like objects containing the blocks within the given StreamField.
+
+##### `set_stream_data(page_or_revision, field_name, stream_data)`
+
+Set the stream field data for a given field name on a page or a revision. 
+
+`stream_data` must be a list of `dict`-like objects containing the blocks within the given StreamField.
```

Comment:
Should this documentation mention that `set_stream_data` commits the change to the database (e.g. calls `save`)? It might not necessarily be obvious to the caller. Django tends to use an optional `commit` parameter ([example](https://docs.djangoproject.com/en/1.8/topics/forms/modelforms/#the-save-method)), and in the future it might be nice to add this should we want to make multiple changes to a page without saving multiple times.